### DISCUSSION TITLE
トレーシングの機能について

### DISCUSSION THREAD

**Discussion by a8119187** [a8119037/isfw2#6] [Ideas]
## 概要
機能としてこういう感じにしたいってのを適当に上げていく

**Comment by a8119187** [comment]
出席時に、QRコードの情報送ってもらって、座った場所のデータを送ってもらう

**Reply by a8119187** [reply]
QRコードでの実装は後回しで、一回文字列で登録機能を実装する。

**Comment by a8119187** [comment]
接触してる人、接触時間を記録して、芋づる式に接触者を表示できるようにする

**Comment by a8119187** [comment]
誰と誰が密な関係か可視化する

**Comment by a8119037** [comment]
- 座席にQRコードを貼り、記録する。
- 体温を記入してもらう。
- 近くに座っている人とそうでない人を可視化する。（ネットワークのような感じ）
- コロナ陽性者が出た時に、濃厚接触者を判定する。

**Comment by a8119187** [comment]
この人が感染者になったとき、どれだけの人がやばいかみたいなのをシミュレーションできる。

**Comment by a8119187** [comment]
濃厚接触者になったら通知してくれる（**システムの必須要件っぽい**）

**Comment by a8119037** [comment]
- 席が近い人と席が遠い人を利用者に知らせる。
- 長い時間近くにいる人に対して警告する。

**Comment by a8119187** [comment]
感染したらその旨を匿名で入力できる。その情報をもとに濃厚接触者を探し、通知できる

**Reply by a8119187** [reply]
感染してなくても、毎回アンケートとかやって危険度を通知してもいいかもしれない。

**Comment by a8119037** [comment]
座席の距離を数値化するのが少し難しい。

**Comment by a8119037** [comment]
Followsテーブルのような席と席の関係性のテーブルを作る。（距離や配置）

**Comment by a8119187** [comment]
とりあえずこれらを一回まとめて、ある程度のまとまりの機能別にissueに上げて、個別にアサインしていく感じで良さそう

**Comment by a8119187** [comment]
# 一回まとめて、ある程度のまとまりの機能別にissueにして、個別にアサインしていく感じで良さそう

1. 出席時に、QRコードの情報送ってもらって、座った場所のデータを送ってもらう
2. 誰と誰が密な関係か可視化する
3. 接触してる人、接触時間を記録して、芋づる式に接触者を表示できるようにする
4. 座席にQRコードを貼り、記録する。
5. 体温を記入してもらう。
6. 近くに座っている人とそうでない人を可視化する。（ネットワークのような感じ）
7. コロナ陽性者が出た時に、濃厚接触者を判定する。
8. この人が感染者になったとき、どれだけの人がやばいかみたいなのをシミュレーションできる。
9. 接触者になったら通知してくれる（システムの必須要件っぽい）
10. 席が近い人と席が遠い人を利用者に知らせる。
11. 長い時間近くにいる人に対して警告する。
12. 感染したらその旨を匿名で入力できる。その情報をもとに濃厚接触者を探し、通知できる
13. 感染してなくても、毎回アンケートとかやって危険度を通知してもいいかもしれない。
14. 座席の距離を数値化するのが少し難しい。
15. Followsテーブルのような席と席の関係性のテーブルを作る。（距離や配置）

**Comment by a8119187** [comment]
# 優先順位高い順に

- [x] 1. ログイン機能
- [x] 2. フォーム（アカウント登録）でUsersのデータを登録してもらう
- [ ] 3. ログイン時に（？）、出席場所のデータを登録させる
- [ ] 4. ユーザ間の距離を表に出力する
- [ ] 5. 距離が近いユーザをリストにまとめる
- [ ] 6. 陽性者の登録をする
- [ ] 7. 陽性者情報をユーザからは匿名化する
- [ ] 8. 接触者に通知する（ユーザ画面に表示する）
- [ ] 9. 座席にQRコードを貼り、QRで席を登録可能にする
- [ ] 10. ユーザごとの接触時間を表示する
- [ ] 11. アンケート機能を追加して、体温とか体調とかを入力可能にする
- [ ] 12. ユーザごとの感染危険度などを座席の距離、接触時間などから表示する

**Reply by a8119037** [reply]
1.完了

**Reply by a8119187** [reply]
3. ログイン時に（？）、出席場所のデータを登録させる

- **_issue立てた_**
- #11
